examples/quantum1d/1.ising-cft/main.jl

```diff
@@ -51,33 +51,29 @@ either diagramatically as
 
 or in the code as:
 """
-
-id = complex(isomorphism(ℂ^2, ℂ^2))
-@tensor O[-1 -2; -3 -4] := id[-1, -3] * id[-2, -4]
-T = periodic_boundary_conditions(InfiniteMPO([O]), L)
+function O_shift(L)
+    id = complex(isomorphism(ℂ^2, ℂ^2))
+    @tensor O[-1 -2; -3 -4] := id[-1, -3] * id[-2, -4]
+    T = periodic_boundary_conditions(InfiniteMPO([O]), L)
+    return T;
+end
 
 md"""
 We can then calculate the momentum of the groundstate as the expectation value of this
 operator. However, there is a subtlety because of the degeneracies in the energy
 eigenvalues. The eigensolver will find an orthonormal basis within each energy subspace, but
 this basis is not necessarily a basis of eigenstates of the translation operator. In order
 to fix this, we diagonalize the translation operator within each energy subspace.
+The resulting energy levels have one-to-one correspondence to the operators in CFT, where the momentum is related to their conformal spin as $P_n = \frac{2\pi}{L}S_n$.
 """
-
-momentum(ψᵢ, ψⱼ=ψᵢ) = angle(dot(ψᵢ, T * ψⱼ))
-
+    
 function fix_degeneracies(basis)
-    N = zeros(ComplexF64, length(basis), length(basis))
+    L = length(basis[1])
     M = zeros(ComplexF64, length(basis), length(basis))
     for i in eachindex(basis), j in eachindex(basis)
-        N[i, j] = dot(basis[i], basis[j])
-        M[i, j] = momentum(basis[i], basis[j])
+        M[i, j] = dot(basis[i],O_shift(L)*basis[j])
```

Comment:
probably want to take the `O_shift` constructor out of the loop here to avoid having to recompute it every time.